### PR TITLE
Delete MoveIt fake_controller_manager

### PR DESCRIPTION
**Pull request by AndyZe**
This deletes the `fake_controller_manager` plugin. I don't believe it's needed anymore because ros2_control supports simulation in a fairly easy way. All you need to do is specify fake components in `your_robot.ros2_control.xacro`. ([example](https://github.com/ros-planning/moveit_resources/blob/ros2/panda_moveit_config/config/panda.ros2_control.xacro))

```
<hardware>
    <plugin>fake_components/GenericSystem</plugin>
</hardware>
```

Then the fake controllers get launched just like a real hardware controller from a launch file:

> ros2_controllers_path = os.path.join(
>     get_package_share_directory("moveit_resources_panda_moveit_config"),
>    "config",
>    "panda_ros_controllers.yaml",
> )
> ros2_control_node = Node(
>     package="controller_manager",
>     executable="ros2_control_node",
>     parameters=[robot_description, ros2_controllers_path],
> )

This is great! It's only a few words difference between hardware and simulation.

I also think it would be good to delete this from MoveIt2 sooner rather than later, to minimize the number of people using fake_controller_manager.

Tagging @v4hn and @rhaschke because they were the code owners in MoveIt1.

### CODE REVIEW

**Review by henningkayser — APPROVED** [review]
I fully agree. Making ros2_control GenericSystem the default minimizes the changes for running setups on hardware and it directly supports all kinds of controllers. I also don't see the use case for the fake controller anymore. I'd still wait for a second opinion by either one of the code owners.

File: .github/CODEOWNERS.disabled
Comment (JafarAbdi) [inline .github/CODEOWNERS.disabled:35]: Should we remove `moveit_ros_control_interface` .? we're not using it at all

**Review by tylerjw — APPROVED** [review]
This needs to be rebased for us to merge it.

File: .github/CODEOWNERS.disabled
Comment (AndyZe) [reply .github/CODEOWNERS.disabled:35]: How about in a new PR?

**Pull request merged**
